core/vm/instructions.go

```diff
@@ -514,25 +513,18 @@ func opCreate(instr instruction, pc *uint64, env Environment, contract *Contract
 		offset, size = stack.pop(), stack.pop()
 		input        = memory.Get(offset.Int64(), size.Int64())
 		gas          = new(big.Int).Set(contract.Gas)
-		addr         common.Address
-		ret          []byte
-		suberr       error
 	)
-
 	contract.UseGas(contract.Gas)
-	ret, addr, suberr = env.Create(contract, input, gas, contract.Price, value)
-	if suberr != nil {
+	_, addr, suberr := env.Create(contract, input, gas, contract.Price, value)
+	// Push new item to the stack based on the returned error (and phase)
+	// If everything was successfull and no error was returned the address
+	// of the created contract is pushed to the stack, otherwise 0.
+	if params.IsHomestead(env.BlockNumber()) && suberr == CodeStoreOutOfGasError {
+		stack.push(new(big.Int))
+	} else if suberr != nil && suberr != CodeStoreOutOfGasError {
```

Comment:
These two conditions leak out an error combo (**not** homestead **but** err **equals** CodeStoreOutOfGasError) into to success branch. I'm not sure based on the PR whether we only return the OutOfGasError in homestead, but these ifs don't fully cover the input space and will be very hard to realize in a later refactor that this is intended or a bug, so my suggestion is:

1. if out of gas is allowed in frontier, then make sure the error paths cover it
2. if the current code is indeed the one desired, then I would separate out this special case into its own branch with a comment as to why it's explicitly needed.

```go
switch err {
  case nil:
    stack.push(addr.Big())

  case suberr == CodeStoreOutOfGasError:
    if params.IsHomestead(env.BlockNumber()) {
      stack.push(new(big.Int))
    } else {
      // Special case because... xyz
      stack.push(addr.Big())
    }

  default:
    stack.push(new(big.Int))
}
```